src/components/PlaylistCard/PlaylistCard.jsx

```diff
@@ -0,0 +1,40 @@
+import axios from "axios";
+import { useToast } from "../../customHooks";
+import { useState } from "react";
+import { usePlaylistContext } from "../../context";
+import { Link } from "react-router-dom";
+
+export const PlaylistCard = ({ id, title, src, alt, videoCount }) => {
+    //MODAL -> This needs to be moved(refactored)
+    const [modalVisible, setModalVisible] = useState(false);
+    const { showToast } = useToast();
+
+    const openModal = () => {
+        setModalVisible(true)
+    }
+
+    const closeModal = () => {
+        setModalVisible(false)
+    }
```

Comment:
Instead of writing separate function, this can be used as online line arrow function directly

`onClick={ () => setModalVisible(true} }`